src/main/sensors/esc_sensor.c

```diff
@@ -96,71 +97,79 @@ typedef enum {
 static bool tlmFrameDone = false;
 static uint8_t tlm[ESC_SENSOR_BUFFSIZE] = { 0, };
 static uint8_t tlmFramePosition = 0;
+
 static serialPort_t *escSensorPort = NULL;
+
 static escSensorData_t escSensorData[MAX_SUPPORTED_MOTORS];
-static uint32_t escTriggerTimestamp = -1;
-static uint32_t escLastResponseTimestamp;
-static uint8_t timeoutRetryCount = 0;
-static uint8_t totalRetryCount = 0;
 
+static escSensorTriggerState_t escSensorTriggerState = ESC_SENSOR_TRIGGER_STARTUP;
+static uint32_t escTriggerTimestamp;
 static uint8_t escSensorMotor = 0;      // motor index
-static bool escSensorEnabled = false;
-static escSensorTriggerState_t escSensorTriggerState = ESC_SENSOR_TRIGGER_WAIT;
 
-static void escSensorDataReceive(uint16_t c);
-static uint8_t update_crc8(uint8_t crc, uint8_t crc_seed);
-static uint8_t get_crc8(uint8_t *Buf, uint8_t BufLen);
-static void selectNextMotor(void);
+static escSensorData_t combinedEscSensorData;
+static bool combinedDataneedsUpdate = true;
```

Comment:
Just a coding style for readability
combinedDataneedsUpdate should be combinedDataNeedsUpdate